views/js/runner/plugins/navigation/next.js

```diff
@@ -96,41 +101,85 @@ define([
-                testRunner.next();
-            }
+            var testData = testRunner.getTestData();
+            var testConfig = testData.config || {};
+            var pluginShortcuts = (testConfig.shortcuts || {})[this.getName()] || {};
 
             //create the button (detached)
             this.$element = createElement(testRunner.getTestContext());
 
             //plugin behavior
-            this.$element.on('click', function(e){
-                var enable = _.bind(self.enable, self);
+            /**
+             * @param {Boolean} nextItemWarning - enable the display of a warning when going to the next item.
+             * Note: the actual display of the warning depends on other conditions (see nextWarningHelper)
+             */
+            function doNext(nextItemWarning) {
                 var context = testRunner.getTestContext();
+                var map = testRunner.getTestMap();
+                var nextItemPosition = context.itemPosition + 1;
+
+                var warningHelper = nextWarningHelper({
+                    endTestWarning:     context.options.endTestWarning,
+                    isLast:             context.isLast,
+                    isLinear:           context.isLinear,
+                    nextItemWarning:    nextItemWarning,
+                    nextPart:           mapHelper.getItemPart(map, nextItemPosition),
+                    remainingAttempts:  context.remainingAttempts,
+                    testPartId:         context.testPartId
+                });
 
-                e.preventDefault();
+                function enable() {
+                    testRunner.trigger('enablenav enabletools');
+                }
 
-                if(self.getState('enabled') !== false){
-                    self.disable();
-                    if(context.options.endTestWarning && context.isLast){
+                if(self.getState('enabled') !== false) {
+                    testRunner.trigger('disablenav disabletools');
+
+                    if (warningHelper.shouldWarnBeforeEnd()) {
                         testRunner.trigger(
                             'confirm.endTest',
                             messages.getExitMessage(
                                 __('You are about to submit the test. You will not be able to access this test once submitted. Click OK to continue and submit the test.'),
                                 'test', testRunner),
-                            _.partial(doNext, context), // if the test taker accept
+                            _.partial(triggerNextAction, context), // if the test taker accept
+                            enable  // if the test taker refuse
+                        );
+
+                    } else if (warningHelper.shouldWarnBeforeNext()) {
+                        testRunner.trigger(
+                            'confirm.next',
+                            __('You are about to go to the next item. Click OK to continue and go to the next item.'),
```

Comment:
The message is redundant...
What about something like:
"You are about to go to the next item. You may not be able to go back to the current item. Continue anyway ?"